dist/images/Dockerfile.fedora

```diff
@@ -9,25 +9,115 @@
 # are built locally and included in the image (instead of the rpm)
 #
 
-FROM fedora:41
+ARG OVN_FROM=koji
+
+#############################################
+# Stage to get OVN and OVS RPMs from source #
+#############################################
+FROM fedora:41 AS ovnbuilder
 
 USER root
 
 ENV PYTHONDONTWRITEBYTECODE yes
 
+# Install tools that are required for building ovs/ovn.
+RUN INSTALL_PKGS="git rpm-build dnf-plugins-core" && \
+    dnf install --best --refresh -y --setopt=tsflags=nodocs $INSTALL_PKGS
+
+# Clone OVS Source Code.
+ARG OVS_REPO=https://github.com/openvswitch/ovs.git
+ARG OVS_GITREF=*branch-*
+WORKDIR /root
+RUN mkdir ovs && pushd ovs && \
+    git init && \
+    git remote add origin $OVS_REPO && \
+    git fetch origin $(git ls-remote $OVS_REPO "$OVS_GITREF" | awk '{print $2}' | sed "s#refs/.*/##g" |sort -V | tail -1) --depth 1 && \
+    git reset --hard FETCH_HEAD && \
+    popd
+
+# Build OVS rpms.
+WORKDIR /root/ovs
+RUN sed -e 's/@VERSION@/0.0.1/' rhel/openvswitch-fedora.spec.in > /tmp/ovs.spec
+RUN dnf builddep -y /tmp/ovs.spec
+RUN ./boot.sh
+RUN ./configure
+RUN make -j rpm-fedora
+RUN rm rpm/rpmbuild/RPMS/x86_64/*debug*
+RUN rm rpm/rpmbuild/RPMS/x86_64/*devel*
+RUN git log -n 1
+
+# Clone OVN Source Code.
+ARG OVN_REPO=https://github.com/ovn-org/ovn.git
+ARG OVN_GITREF=main
+WORKDIR /root
+RUN mkdir ovn && pushd ovn && \
+    git init && \
+    git remote add origin $OVN_REPO && \
+    git fetch origin $(git ls-remote $OVN_REPO "$OVN_GITREF" | awk '{print $2}' | sed "s#refs/.*/##g" |sort -V | tail -1) --depth 1 && \
```

Comment:
I am not clear on what you are achieving with `git ls-remote`. 

You mentioned
```
Going to use a version of this that do not need to clone the repo
```
What clone of the repo are you referring to?

Reply:
The dumitru's proposal was 
```
        git checkout \
          $(git branch -a -l '*branch-*' | sed 's/remotes\/origin\///' | \
            sort -V | tail -1)
```

For that you need to clone the repo, with ls-remove you can even do that out of tree.

Reply:
Dumitru's proposal was for OVS, not for OVN?

Reply:
And specifically for OVS, if we decide to go for a stable release, why wouldn't we just download a built release instead of building ourselves?

Reply:
Is a good feature for OVN too, so I have replicate for it here.

Reply:
> Dumitru's proposal was for OVS, not for OVN?

Is a good feature for OVN too, so I have replicate it.

> And specifically for OVS, if we decide to go for a stable release, why wouldn't we just download a built release instead of building ourselves?

@dceara how often are you going to need to test ovn-kubernetes with custom OVS ?

Reply:
A good feature for what? But you are trolling me again :P I see now that you are making a meaningful use of it in the new `dist/images/fedora-build-args.sh`

Reply:
> A good feature for what? But you are trolling me again :P I see now that you are making a meaningful use of it in the new `dist/images/fedora-build-args.sh`

Like using "branch-*" to have latest stable OVN branch

Reply:
Ok now I get it